documentation/design_docs/design/resource-definitions.adoc

```diff
@@ -0,0 +1,99 @@
+== why
+
+The address controller and standard controller needs a way to tie plans into templates that get
+instantiated. Previously, the plans have contained this information directly, and a 'special' name
+was used for variations like pooled queues.
+
+With the new plans, resources are named along with an amount of resource required for that plan.
+This proposal defines a format for such resources, which makes them configurable for the messaging
+service admin.
+
+Essentially, a resource definition ties a template name together with parameters.
+
+== what/how
+
+ConfigMaps would be used to store the resource definition. A type label is used to distinguish it
+from other configmaps. In the global namespace, all resources referenced by the address space plans
+and address plans need to exist.
+
+A resource definition would be specified in a configmap containing the actual definition in one of
+the fields. The fields of a resource definition is:
+
+name [required], name of the resource
+
+template [optional], name of the template to instantiate
+
+template [optional], parameters to pass to the template
+
+An example resource definition:
+
+```
+{
+    "apiVersion": "enmasse.io/v1",
+    "kind": "ResourceDefinition",
+    "metadata": {
+        "name": 
+    },
+    "template": "persisted-broker",
+    "parameters": [
+        {
+            "name": "MY_PARAMETER",
+            "value": "val1"
+        }
+    ]
+}
+```
+
+The resource definition configmap would look like this (in YAML for readability)
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: resource-definition-foo
+  labels:
+    type: resource-definition-config
+data:
+  definition: |-
+    {
+        "apiVersion": "enmasse.io/v1",
+        "kind": "ResourceDefinition",
+        "metadata": {
+            "name": 
+        },
+        "template": "persisted-broker",
+        "parameters": [
+            {
+                "name": "MY_PARAMETER",
+                "value": "val1"
+            }
+        ]
+    }
+```
+
+All resource definitions are stored and validated globally by the address controller. When an
+address space is created, the resource definitions are copied into the address space.
+
+The ResourceDefinition can be referenced by both AddressPlans and AddressSpacePlans. The plans are
```

Comment:
Perhaps better to say 'A', rather than 'The' here? The point being that it sis unlikely you would ever refer to the same resource definition from AddressPlan and AddressSpacePlan.